state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
 > A response with an empty `payload` and empty `enrs` indicates that the node is not aware of any closer nodes, *nor* does the node have the requested content.
 
 
-#### Advertise (0x07)
+#### Offer (0x07)
 
-Advertise a set of content keys that this node has proofs available for.
+Offer a set of content keys that this node has proofs available for.
 
 ```
 message_id := 7
 type       := request
-sedes      := List[byte_list, max_length=32]
+sedes      := List[byte_list, max_length=64]
 ```
 
 The payload of this message is a list of encoded `content_key` entries.
 
 
-#### RequestProofs (0x08)
+#### Accept (0x08)
 
-Response message to Advertise (0x07).
+Response message to Offer (0x07).
 
-Request the proofs for a piece of content.
+Signals interest in receiving the offered data from the corresponding Offer message.
 
-> Despite the name of this message having the name "Request" in it, this is a response message.
 
 ```
 message_id := 8
 type       := response
-sedes      := Container(connection_id: bytes4, content_keys: List[byte_list, max_length=32]]
+sedes      := Container(connection_id: Bytes2, content_keys: BitList[max_length=64]]
 ```
 
 * `connection_id`: ConnectionID to be used for a uTP stream
     * ConnectionID values should be randomly generated.
-* `content_keys`: The set of encoded content keys that should be sent.
-    * Only content keys from the corresponding Advertise message are valid.
+* `content_keys`: Signals which content keys are desired
+    * A bit-list corresponding to the offered keys with the bits in the positions of the desired keys set to `1`.
 
 Upon *sending* this message, the requesting node should *listen* for an incoming uTP stream with the generated `connection_id`.
 
 Upon *receiving* this message, the serving node should initiate a uTP stream.
 
-Proofs sent across the stream should be framed with a 4-byte length prefix.
+> TODO: how does message framing across the stream work for individual messages.
+
+
+## Gossip
+
+The state network will use a multi stage mechanism to store new and updated state data.
+
+### Overview
+
+The state data is stored in the network in two formats.
+
+- A: Trie nodes
+    - Individual leaf and intermediate trie nodes.
+    - Gossiped with a proof against a recent state root
+    - Stored without a proof not anchored to a specific state root
```

Comment:
No, the current plan is for nodes to 1) occasionally pick the least recently validated node they are storing (which includes the path data) and to walk the trie to see if the node is still valid and 2) to gossip the exclusion proof they build in this scanning to other nodes so that they can delete them without needing to do the walk.